birdhouse/optional-components/prometheus-log-parser/config/thredds/prometheus-log-exporter.py

```diff
@@ -54,13 +54,16 @@
 def parse_line(line):
     match = REGEX.match(line)
     if match:
-        labels = {label: match.group(label) or "" for label in LABEL_KEYS}
+        labels = {k: match.groupdict().get(k, None) for k in LABEL_KEYS}
         # Tweaks
-        if labels["tds_service"] == "dodsC":
-            labels["dataset"] = labels["dataset"].removesuffix(".dods")
-            labels["variable"] = match.group("thredds_request").split(".")[0]
-        elif labels["tds_service"] == "ncss":
-            labels["dataset"] = labels["dataset"].removesuffix("/dataset.html")
+        if labels.get("tds_service") == "dodsC":
+            labels["dataset"] = (labels.get("dataset", "").removesuffix(".dods")
```

Comment:
`"dataset"` will always be in `labels` because of line 57 so the default for `get` will never be used here. If `labels["dataset"]` is `None` then this will raise an error because None types don't have a `removesuffix` method.

I'd recommend keeping the default value in `labels` as the empty string and not bothering with `get` here.